File: erlang-rebar3/hello-binary/src/hello.erl
%% M61 erlang-rebar3/hello-binary fixture: escript main module.
%%
%% Declares ``main/1`` so ``rebar3 escriptize`` can produce a
%% self-contained escript that invokes this callback when run.
%% Uses only Erlang's stdlib (``io:format/1``) so the build needs no
%% external Hex dependencies — keeping the M61 offline-mode contract
%% trivially satisfied.

-module(hello).
-export([main/1]).

main(_Args) ->
    io:format("hello from erlang-rebar3-hello-binary~n").

File: elixir-mix/hello-binary/lib/hello.ex
## M62 elixir-mix/hello-binary fixture: escript main module.
##
## Declares ``main/1`` so ``mix escript.build`` can produce a
## self-contained escript that invokes this callback when run via
## ``escript hello`` (or the M62 convention's
## ``<root>/.repro/build/hello/hello.cmd`` wrapper on Windows). Uses
## only Elixir's stdlib (``IO.puts/1``) so the build needs no
## external Hex dependencies — keeping the M62 offline-mode contract
## trivially satisfied.

defmodule Hello do
  def main(_args) do
    IO.puts("hello from elixir-mix-hello-binary")
  end
end
